code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -300,6 +300,61 @@
 	. = ..()
 	AddComponent(/datum/component/armor_plate)
 
+
+//Nanotrasen medical Protosuit
+
+/obj/item/clothing/suit/space/hardsuit/protosuit
+	name = "protosuit hardsuit"
+	desc = "A failed NanoTrasen experiment, intended to be a unique medical hardsuit based on stolen Syndicate samples."
+	icon_state = "protosuit"
+	hardsuit_type = "protosuit"
+	armor = list(MELEE = 30, BULLET = 15, LASER = 15, ENERGY = 25, BOMB = 30, BIO = 100, RAD = 25, FIRE = 75, ACID = 75, WOUND = 15)
+	allowed = allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/storage/firstaid, /obj/item/healthanalyzer, /obj/item/stack/medical)
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/protosuit
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit
+	name = "protosuit hardsuit helmet"
+	desc = "A helmet, made in old failed NanoTrasen experiment, which was intended to create a medical hardsuit based of stolen Syndicate ones. You can see a small DNA-sampler on the bottom of it."
+	icon_state = "protosuit"
+	hardsuit_type = "protosuit"
+	armor = list(MELEE = 30, BULLET = 15, LASER = 15, ENERGY = 25, BOMB = 30, BIO = 100, RAD = 25, FIRE = 75, ACID = 75, WOUND = 15)
+	light_range = 7
+	resistance_flags = FIRE_PROOF
+
+	var/initialized = FALSE
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(slot != ITEM_SLOT_HEAD)
+		if(suit)
+			if(!user.dna)
+				display_visor_message("<span class='notice'>Initialization error. Unable to recognise user. Halting setup process.</span>")
+				return
+
+			var/datum/data/record/R = find_record("name", user.dna.real_name, GLOB.data_core.medical) //The helmet is based on DNA
+			if(!R)
+				display_visor_message("<span class='notice'>Initialization error. Unable to recognise user. Halting setup process.</span>")
+				return
+
+			display_visor_message("<span class='notice'>Initialization complete. User recognised. Hello, [user.dna.real_name].</span>")
+			ADD_TRAIT(user, TRAIT_SELF_AWARE, CLOTHING_TRAIT)
+			ADD_TRAIT(user, TRAIT_MEDICAL_HUD, CLOTHING_TRAIT)
+			initialized = TRUE
+	else
+		if(initialized)
+			REMOVE_TRAIT(user, TRAIT_SELF_AWARE, CLOTHING_TRAIT)
+			REMOVE_TRAIT(user, TRAIT_MEDICAL_HUD, CLOTHING_TRAIT)
+			display_visor_message("<span class='notice'>De-initializing HUD system. Have a great shift.</span>")
+			initialized = FALSE
+
+/obj/item/clothing/head/helmet/space/hardsuit/protosuit/dropped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(initialized)
```

Comment:
why is this var needed

Reply:
To know should you display the exit message or not

Reply:
it should always display it?